### PR TITLE
Update limiter function to work on both 2D and 2D x 1D extruded spaces

### LINKED ISSUE #584: Extend limiters to work on both 2D and hybrid 2D x 1D spaces
For now the 1D x 1D extruded space support was not deemed a priority. We can consider its addition later on. 

Need to: 

- [x] Extend `src/Limiters/limiter.jl` function to work equally on 2D and hybrid 2D x 1D spaces
- [x] Add  2D x 1D space unit test

### PR DESCRIPTION
**Pull request by valeriabarra**
This PR updates the limiter function to work on both 2D and 2D x 1D extruded spaces. 

For now the 1D x 1D extruded space support was not deemed a priority and can be added in a following PR with related tests/examples. 

This will close #584  

PS: I also took the occasion to rename `examples/sphere/opt_limiters_solidbody.jl` -> `examples/sphere/limiters_advection.jl` for consistency with the plane and following examples I am working on to address #575 .

### CODE REVIEW

**Comment by valeriabarra** [comment]
bors try

**Comment by valeriabarra** [comment]
bors try

**Comment by valeriabarra** [comment]
Unit tests for the 2D cases work with the extended interface. 
Unit test for 2D + 1D hybrid domain does not work at the moment (blocking issue opened #586 )

File: src/Limiters/limiter.jl [hunk 1]
```diff
@@ -29,99 +29,110 @@ function quasimonotone_limiter!(
     max_ρq;
     rtol,
 )
-    space = axes(ρ)
+    if typeof(size(min_ρq)) == Tuple{Int}
```
Comment (charleskawczynski) [inline src/Limiters/limiter.jl:32]:
```suggestion
    if ndims(min_ρq) == 1
```
I'm wondering if this is simpler

File: src/Limiters/limiter.jl [hunk 2]
```diff
@@ -29,99 +29,110 @@ function quasimonotone_limiter!(
...
+            ρq_e_data = Fields.field_values(ρq_e_slab)
+            # Compute ρ's and ρq's masses over an element he
+            ρ_e_mass = sum(ρ_e_slab)
+            ρq_e_mass = sum(ρq_e_slab)
 
-        q_data .= ρq_e_data ./ ρ_e_data
+            # This should never happen, but if it does, don't limit
+            if ρ_e_mass <= zero(FT)
+                error("Negative elemental mass!")
+                continue
```
Comment (charleskawczynski) [inline src/Limiters/limiter.jl:60]: What's the `continue` for?-- it'll never be executed

File: src/Limiters/limiter.jl [hunk 3]
```diff
@@ -29,99 +29,110 @@ function quasimonotone_limiter!(
...
-            if mass_change > 0
+            # Weighted least squares problem iteration loop
+            for iter in 1:(Nq * Nq - 1)
+                mass_change = zero(FT)
                 # Iterate over quadrature points
                 for j in 1:Nq, i in 1:Nq
-                    if q_data[i, j] < max_ρq[e]
-                        weights_sum += ρ_wJ_data[i, j]
+                    ρ_wJ_data[i, j] =
+                        local_geometry_slab[i, j].WJ .* ρ_e_data[i, j]
```
Comment (charleskawczynski) [inline src/Limiters/limiter.jl:84]: What's the shape of `ρ_e_data[i, j]`? Do we need to dot operation here?

File: src/Limiters/limiter.jl (hunk 1 above)
Comment (valeriabarra) [reply src/Limiters/limiter.jl:32]: Yes, thank you!

File: src/Limiters/limiter.jl (hunk 2 above)
Comment (valeriabarra) [reply src/Limiters/limiter.jl:60]: That's right, thanks for catching that! It was accidentally left behind. It was there before I added the `error`, as suggested in the PR's review  when this was added

File: src/Limiters/limiter.jl (hunk 3 above)
Comment (valeriabarra) [reply src/Limiters/limiter.jl:84]: Oh, thanks for spotting this, too! It is absolutely not necessary, and in fact, we do not use it in the analog following lines 89 and 93 below. It was just there by accident

**Comment by valeriabarra** [comment]
bors try

**Comment by valeriabarra** [comment]
Hey @charleskawczynski , this PR is ready on my hand. I'd appreciate a final look. Just to summarize, this PR: 

- renames examples/sphere/opt_limiters_solidbody.jl -> examples/sphere/limiters_advection.jl (for consistency with the plane and following examples I am working on to address #575 ).
- extends Limiters function to work both on 2D and 2D x 1D extruded spaces, and adds unit tests for the latter

Thanks!

**Comment by valeriabarra** [comment]
I went ahead and squashed commits

File: src/Limiters/limiter.jl [hunk 4]
```diff
@@ -29,99 +29,113 @@ function quasimonotone_limiter!(
...
+        ρq_level_mass = sum(Fields.slab(ρq, ve))
+        # On each level, traverse horizontal elements
+        for he in 1:horz_n_elems
+            ρ_he_slab = Fields.slab(ρ, ve, he)
+            ρq_he_slab = Fields.slab(ρq, ve, he)
+            ρ_he_data = Fields.field_values(ρ_he_slab)
+            ρq_he_data = Fields.field_values(ρq_he_slab)
+            # Compute ρ's and ρq's masses over an horizontal element e
+            ρ_he_mass = sum(ρ_he_slab)
+            ρq_he_mass = sum(ρq_he_slab)
```
Comment (charleskawczynski) [inline src/Limiters/limiter.jl:59]: This sum is over a single vertical element?

**Review by simonbyrne — APPROVED** [review]

**Pull request merged**